server/src/main/java/org/opensearch/cluster/routing/allocation/decider/ConcurrentRecoveriesAllocationDecider.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/*
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.cluster.routing.allocation.decider;
+
+import org.opensearch.cluster.routing.RoutingNode;
+import org.opensearch.cluster.routing.ShardRouting;
+import org.opensearch.cluster.routing.allocation.RoutingAllocation;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Setting.Property;
+import org.opensearch.common.settings.Settings;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * This {@link AllocationDecider} controls the number of currently in-progress
+ * re-balance (relocation) operations and restricts node allocations if the
+ * configured threshold is reached.
+ * <p>
+ * Re-balance operations can be controlled in real-time via the cluster update API using
+ * <code>cluster.routing.allocation.cluster_concurrent_recoveries</code>. Iff this
+ * setting is set to <code>-1</code> the number of cluster concurrent recoveries operations
+ * are unlimited.
+ */
+public class ConcurrentRecoveriesAllocationDecider extends AllocationDecider {
+
+    public static final String NAME = "cluster_concurrent_recoveries";
+
+    protected final Logger logger;
+
+    public static final Setting<Integer> CLUSTER_ROUTING_ALLOCATION_CLUSTER_CONCURRENT_RECOVERIES_SETTING =
+        Setting.intSetting("cluster.routing.allocation.cluster_concurrent_recoveries", -1, -1,
+            Property.Dynamic, Property.NodeScope);
+
+    private volatile int clusterConcurrentRecoveries;
+
+    public ConcurrentRecoveriesAllocationDecider(Settings settings, ClusterSettings clusterSettings) {
+        super();
+        this.logger = LogManager.getLogger(getClass());
+        this.clusterConcurrentRecoveries = CLUSTER_ROUTING_ALLOCATION_CLUSTER_CONCURRENT_RECOVERIES_SETTING.get(settings);
+        logger.debug("using [cluster_concurrent_rebalance] with [{}]", clusterConcurrentRecoveries);
```

Comment:
Lets remove this since this is logged at node startup anyways?

Reply:
Keeping it for now as consistent with other deciders like ConcurrentRebalanceAllocationDecider